### PR TITLE
Add NumberInput

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

File: src/components/NumberInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import InputContainer, { InputContainerProps } from '../InputContainer';
+import RawInput, { RawInputProps } from '../RawInput';
+
+export type NumberInputProps<T> = Omit<InputContainerProps, 'input'> & RawInputProps<T>;
+
+function NumberInput<T extends string>(props: NumberInputProps<T>) {
```
Comment (tnagorra) [inline src/components/NumberInput/index.tsx:18]: The value for number input should be either `number` or `undefined`

File: src/components/NumberInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        labelContainerClassName,
+        readOnly,
+        uiMode,
+        onChange,
+        name,
+        value,
+        ...rawInputProps
+    } = props;
+
+    const handleChange = (v: string, n: T, e: React.FormEvent<HTMLInputElement>) => {
```
Comment (frozenhelium) [inline src/components/NumberInput/index.tsx]: use `React.useCallback`

**Review by frozenhelium — APPROVED** [review]

File: src/components/NumberInput/index.tsx (hunk 1 above)
Comment (frozenhelium) [reply src/components/NumberInput/index.tsx:18]: `T` is for `name` not for value in this case

**Review by frozenhelium — APPROVED** [review]

File: src/components/NumberInput/index.tsx (hunk 1 above)
Comment (tnagorra) [reply src/components/NumberInput/index.tsx:18]: Not talking about the T for `name` here!

File: src/stories/NumberInput.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+const Template = (args: NumberInputProps) => {
+    const [{ value }, updateArgs] = useArgs();
+
+    const handleChange = (e) => {
+        updateArgs({ value: e });
+    };
+
+    console.log('outside', value);
```
Comment (frozenhelium) [inline src/stories/NumberInput.stories.tsx]: We should remove this may be?

**Review by frozenhelium — APPROVED** [review]

File: src/stories/NumberInput.stories.tsx (hunk 3 above)
Comment (tnagorra) [reply src/stories/NumberInput.stories.tsx]: https://www.youtube.com/watch?v=S9RVS8cjNN0

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**